SharpSploit/Execution/Injection/Execution.cs

```diff
@@ -0,0 +1,267 @@
+            }
+            // If there is no such method
+            catch (ArgumentNullException)
+            {
+                throw new PayloadTypeNotSupported(Payload.GetType());
+            }
+        }
+    }
+
+    
+    /// <summary>
+    /// Executes a payload in a remote process by creating a new thread. Allows the user to specify which API call to use for remote thread creation.
+    /// </summary>
+    public class RemoteThreadCreate : ExecutionTechnique
+    {
+        // Publically accessible options
+        public bool suspended = false;
+        public APIS api = APIS.NtCreateThreadEx;
+
+        public enum APIS : int
+        {
+            NtCreateThreadEx = 0,
+            // NtCreateThread = 1, // Not implemented
+            RtlCreateUserThread = 2,
+            CreateRemoteThread = 3
+        };
+
+        // Handle of the new thread. Only valid after the thread has been created.
+        public IntPtr handle = IntPtr.Zero;
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        public RemoteThreadCreate()
+        {
+            DefineSupportedPayloadTypes();
+        }
+
+        /// <summary>
+        /// Constructor allowing options as arguments.
+        /// </summary>
+        public RemoteThreadCreate(bool susp = false, APIS varAPI = APIS.NtCreateThreadEx)
+        {
+            DefineSupportedPayloadTypes();
+            suspended = susp;
+            api = varAPI;
+        }
+
+        /// <summary>
+        /// States whether the payload is supported.
+        /// </summary>
+        /// <author>The Wover (@TheRealWover)</author>
+        /// <param name="Payload">Payload that will be allocated.</param>
+        /// <returns></returns>
+        public override bool IsSupportedPayloadType(PayloadType Payload)
+        {
+            return supportedPayloads.Contains(Payload.GetType());
+        }
+
+        /// <summary>
+        /// Internal method for setting the supported payload types. Used in constructors.
+        /// Update when new types of payloads are added.
+        /// </summary>
+        /// <author>The Wover (@TheRealWover)</author>
+        internal override void DefineSupportedPayloadTypes()
+        {
+            // Defines the set of supported payload types.
+            supportedPayloads = new Type[] {
+                typeof(PICPayload)
+            };
+        }
+
+        /// <summary>
+        /// Only ever called if the user passed in a Payload type without an Inject overload.
+        /// </summary>
+        /// <author>The Wover (@TheRealWover)</author>
+        /// <param name="Payload">Payload type.</param>
+        /// <param name="AllocationTechnique">Allocation technique.</param>
+        /// <param name="Process">The target process.</param>
+        /// <returns></returns>
+        public bool Inject(PayloadType Payload, AllocationTechnique AllocationTechnique, Process Process)
```

Comment:
This is the function mentioned about that non-explicitly hides the function inherited from ExecutionTechnique.

I'd like to see if we could remove the non-explicit hide, and ensure this function works even if an object is typed explicitly as RemoteThreadCreate.

Reply:
I will try to find a way to not need this function.